File: smarthome.FrontEnd/Smarthome/src/app/models/device.ts
export enum DeviceType {
    SecurityCamera = 0,
    WindowSensor = 1,
    SmartLamp = 2,
    MotionSensor = 3
}

export const DeviceTypeNames: Record<DeviceType, string> = {
    [DeviceType.SecurityCamera]: 'Security Camera',
    [DeviceType.WindowSensor]: 'Window Sensor',
    [DeviceType.SmartLamp]: 'Smart Lamp',
    [DeviceType.MotionSensor]: 'Motion Sensor'
}



export interface Device {
    deviceId: number;
    name: string;
    deviceType: DeviceType;
    model: string;
    description: string;
    imageUrl: string;
    companyId: number;

    isInterior?: boolean;
    isExterior?: boolean;
    canDetectMotion?: boolean;
    canDetectPerson?: boolean;

    isWindowOpen?: boolean;

    isTurnedOn?: boolean;
}

File: smarthome.Domain/Device.cs
using System.ComponentModel.DataAnnotations;

namespace smarthome.Domain;

public enum DeviceType
{
    SecurityCamera = 0,
    WindowSensor = 1,
    SmartLamp = 2,
    MotionSensor = 3
}

public abstract class Device
{
    [Key]
    public int DeviceId { get; set; }

    [Required]
    public string Name { get; set; }

    [Required]
    public DeviceType DeviceType { get; set; }

    public string Model { get; set; }

    [Required]
    public string Description { get; set; }

    [Required]
    public string ImageUrl { get; set; }

    [Required]
    public int CompanyId { get; set; }

    [Required]
    public Company Company { get; set; }

    public Device() { }
}
